components/social/org.wso2.carbon.social.core/src/main/java/org/wso2/carbon/social/core/ActivityPublisher.java

```diff
@@ -57,9 +57,23 @@ public long publish(String activity) throws SocialActivityException {
 		return publishActivity(jsonObject);
 	}
 
+	public String update(String activity) throws SocialActivityException {
+		JsonObject jsonActivity;
+		try {
+			jsonActivity = (JsonObject) parser.parse(activity);
+		} catch (JsonSyntaxException e) {
+			String message = "Malformed JSON element found: " + e.getMessage();
+			throw new SocialActivityException(message, e);
```

Comment:
Thanks @daneshk   , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/c9d81e53fb6b8db54f26c7587f30433ca96745d5)